File: LanguageCards/Entities/Word.cs
using LanguageCards.Data.Enums;
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LanguageCards.Data.Entities
{
    public class Word
    {
        public int Id { get; set; }

        /// <summary>
        /// Retrieves a string representation of the word
        /// </summary>
        public string Text { get; set; }

        public int LanguageId { get; set; }
        /// <summary>
        /// Retrieves the language of the word
        /// </summary>
        public Language Language { get; set; }

        public int SpeechPartId { get; set; }
        /// <summary>
        /// Retrieves the part of speech
        /// </summary>
        public SpeechPart SpeechPart { get; set; }

        /// <summary>
        /// Represents the definition of the word
        /// </summary>
        public string Definition { get; set; }

        /// <summary>
        /// Represents the translations of the word
        /// </summary>
        public ICollection<Word> Translations { get; set; }

        /// <summary>
        /// Represents an example (or examples) containing this word
        /// </summary>
        public string Example { get; set; }
    }
}

File: LanguageCards.WebApp/ClientApp/app/models/word.ts
export class Word {
    id: number;
    text: string;
    languageId: number;
    language: {
        id: number;
        name: string;
    }
    speechPartId: number;
    speechPart: {
        id: number;
        name: string;
    }
    definition: string;
    translations: Word[];
    example: string;

    constructor(obj: any) {
        this.id = obj.id;
        this.text = obj.text;
        this.languageId = obj.languageId;
        this.language.id = obj.language.id;
        this.language.name = obj.language.name;
        this.speechPartId = obj.speechPartId;
        this.speechPart.id = obj.speechPart.id;
        this.speechPart.name = obj.speechPart.name;
        this.definition = obj.definition;
        this.translations = obj.translations;
        this.example = obj.example;
    }
}
